### PR TITLE
Add Azure ingestion flow to web console

### PR DESCRIPTION
**Pull request by mcbrewster**
Screen Shot 2020-02-28 at 1 03 39 PM
Screen Shot 2020-02-28 at 1 03 30 PM
**Please do not merge until support for azure has been merged**

Adds the ability to ingest data from azure via the web console. There is now a tile that allows the user to access the Azure ingestion flow if druid-azure-extensions is in the load list. The default input type will be azure. The user can then put a can then input data using A URI or a Prefix. the resulting IO config changes will be : 

```
  "ioConfig": {
    "type": "index_parallel",
    "inputSource": {
      "type": "azure",
      "prefixes": [
        "azure://containerName"
      ]
    }
  },
```

or 
```
  "ioConfig": {
    "type": "index_parallel",
    "inputSource": {
      "type": "azure",
      "prefixes": [
        "azure://container-name/path-name.json"
      ]
    }
  },
```
The user can configure the rest of the spec using the normal web console flow. 

Additionally
 azure is now available as an input type in the input menu for all flows
Screen Shot 2020-02-28 at 1 05 59 PM

### CODE REVIEW

**Comment by vogievetsky** [comment]
> Please do not merge until support for azure has been merged

Can you indicate what this is blocked on?

**Comment by mcbrewster** [comment]
I just wasn’t sure if zachs stuff has been merged

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________

**Comment by vogievetsky** [comment]
I think you are referring to https://github.com/apache/druid/pull/9394 and it is merged!

File: web-console/src/utils/ingestion-spec.tsx [hunk 1]
```diff
@@ -2615,6 +2687,10 @@ export function upgradeSpec(spec: any): any {
         deepSet(spec, 'ioConfig.firehose.type', 's3');
         break;
 
+      case 'static-azure':
+        deepSet(spec, 'ioConfig.firehose.type', 'azure');
+        break;
```
Comment (vogievetsky) [inline web-console/src/utils/ingestion-spec.tsx]: There is no need for this upgrade spec clause, there was never a `'static-azure'`. This is back compat code but Azure ingestion did not used to exist. You can just remove this change.

File: web-console/src/utils/ingestion-spec.tsx [hunk 2]
```diff
@@ -2653,6 +2729,10 @@ export function downgradeSpec(spec: any): any {
         deepSet(spec, 'ioConfig.firehose.type', 'static-s3');
         break;
 
+      case 'azure':
+        deepSet(spec, 'ioConfig.firehose.type', 'static-azure');
+        break;
```
Comment (vogievetsky) [inline web-console/src/utils/ingestion-spec.tsx]: Same story as above.

**Comment by vogievetsky** [comment]
Looks great thank you 👍

**Pull request merged**